src/libs/actions/IOU.js

```diff
@@ -720,11 +720,152 @@ function payIOUReport({
     return promiseWithHandlers;
 }
 
+/**
+ * @param {String} chatReportID - Report ID of the chat where the IOU is.
+ * @param {Number} amount - IOU amount in cents.
+ * @param {String} currency - IOU currency.
+ * @param {String} comment
+ * @param {Array} participants - Participants on the IOU
+ */
+function sendMoneyWithWallet(chatReportID, amount, currency, comment, participants) {
+    const chatReport = _.empty(chatReportID) ? chatReports[`${ONYXKEYS.COLLECTION.REPORT}${chatReportID}`] : ReportUtils.buildOptimisticChatReport(participants);
+
+    const ownerEmail = participants[0].login;
+    const userEmail = participants[1].login;
+    const optimisticIOUReport = ReportUtils.buildOptimisticIOUReport(
+        ownerEmail,
+        userEmail,
+        amount,
+        chatReport.chatReportID,
+        currency,
+        preferredLocale
+    );
+
+    const newSequenceNumber = Report.getMaxSequenceNumber(chatReport.reportID) + 1;
+    const optimisticIouReportAction = ReportUtils.buildOptimisticIOUReportAction(
+        newSequenceNumber,
+        CONST.IOU.REPORT_ACTION_TYPE.PAY,
+        amount,
+        currency,
+        comment,
+        participants,
+        optimisticIOUReport.reportID,
+    );
+
+    API.write('SendMoneyWithWallet',
+        {
+            params,
+        },
+        {
+            optimisticData: [
+                {
+                    onyxMethod: CONST.ONYX.METHOD.MERGE,
+                    key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${chatReport.chatReportID}`,
+                    value: {
+                        [optimisticIouReportAction.reportActionID]: {
+                            optimisticIouReportAction,
+                            // Set pendingAction state for pattern B
```

Comment:
I would move the optimisticData, failureData before this, define it like it is defined here https://github.com/Expensify/App/blob/2356010ff7817ae27e603558b6905474929465d6/src/libs/actions/IOU.js#L580-L626

It makes it cleaner I think